docs/language-service.md

```diff
@@ -0,0 +1,38 @@
+# Key features of the Bicep language service
+
+The bicep language service (implemented in the [bicep VS Code extension](./installing.md#install-the-bicep-vs-code-extension)) is capable of many of the features you would expect out of other language tooling. Here's all of the features that are currently implemented.
+
+## Validation and type checking
+
+The bicep compiler has a type system that ensure your code is authored correctly. We will check the basic types that are built into the bicep language, as well as types of returned functions, and types of resources. Depending on the type of validation, you will see either a warning in yellow which will successfully compile with `bicep build` or you will see an error in red which will fail to compile.
+
+## Intellisense
+
+* code completions
+  * dot-property access
+    * works on any `param` or `var` object, `resource` or `module`
+  * resource property names & property values
+    * works for every `resource` and `module` (e.g. `myModule.outputs.`)
+  * resource types
+
+### Snippets
+
+## Code navigation
+
+### Go to definition
+
+### Find all references
+
+## Refactoring
+
+### Rename symbol
+
+### Formatting
+
+* `ctrl` + `shift` + `f` (windows)
+* supports `.editorconfig`
+* sets its own setting in the `.vscode` directory
```

Comment:
I don't think either of these points are true. We don't support .editorconfig in formatting, and don't update any actual settings in `.vscode`

Reply:
We don't support `.editorconfig` directly because we don't read it, but the settings like `indent_size` and `indent_style` should work as long as the EditorConfig for VSCode extension is installed.

Reply:
I think we should point that out explicitly so people don't get confused.

Reply:
We should probably also point out that the Bicep VSCode extension will set the following default settings once installed:
```
  "[bicep]": {
    "editor.tabSize": 2,
    "editor.insertSpaces": true
  }
```

Users can change the default settings in the following places (sorted by precedence in ascending order):
- VSCode global user settings
- VSCode workspace settings
- `.editorconfig` files (requires `EditorConfig for VSCode` extension to be installed)

Reply:
where are the default settings saved? and presumably, we are only setting these default settings for .bicep files - is that correct?

Reply:
They are saved in the VSCode global default settings (can be viewed via F1 / Ctrl + Shift + P -> Preferences: Open Default Settings). And yes, the settings are only set for .bicep files.

Reply:
Are we documenting the language service here, or the VSCode extension? I assumed the former because of the name of this document. If we're documenting the language service, does it make sense to talk about VSCode settings/extensions?

Reply:
I feel like, at least for now, they should be the same doc since everyone will be experiencing it through the VS Code extension. But to your point, probably should call out the VS Code specific stuff separately.